src/Elastic.Documentation.Site/Navigation/INavigationHtmlWriter.cs

```diff
@@ -10,11 +10,30 @@ public interface INavigationHtmlWriter
 {
 	const int AllLevels = -1;
 
-	Task<string> RenderNavigation(IRootNavigationItem<INavigationModel, INavigationItem> currentRootNavigation, Uri navigationSource, int maxLevel, Cancel ctx = default);
+	Task<INavigationRenderResult> RenderNavigation(IRootNavigationItem<INavigationModel, INavigationItem> currentRootNavigation, Uri navigationSource,
+		int maxLevel, Cancel ctx = default);
 
 	async Task<string> Render(NavigationViewModel model, Cancel ctx)
 	{
 		var slice = _TocTree.Create(model);
 		return await slice.RenderAsync(cancellationToken: ctx);
 	}
 }
+
+public interface INavigationRenderResult
+{
+	string Html { get; init; }
+	string Id { get; init; }
+}
+
+public record OkNavigationRenderResult : INavigationRenderResult
+{
+	public required string Html { get; init; }
+	public required string Id { get; init; }
+}
+
+public record EmptyNavigationRenderResult : INavigationRenderResult
```

Comment:
This can be a static property on `OkNavigationRenderResult` ? 

e.g `OkNavigationRenderResult.Empty` then we can just have a single record `NavigationRenderResult` without an interface.

Reply:
https://github.com/elastic/docs-builder/pull/1514/commits/64966753aa8eb7150b3cedb9e43f2ea4e556b306